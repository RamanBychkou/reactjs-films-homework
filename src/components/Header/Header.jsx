import React from 'react';
import PropTypes from 'prop-types';
import style from './Header.scss';

class Header extends React.Component {
  search = (e) => {
    const { error, clearError, history } = this.props;
    e.preventDefault();
    if (error !== undefined && error !== '') {
      clearError('');
    }
    const query = e.target.children[0].value;
    history.location.pathname = '';
    history.push(`/search/${query}`);
    // return fetchSearchResults(query);
  };

  render() {
    return (
      <header className={style.header}>
        <div className={style.container}>
          <h1 className={style.h1}>FILMS</h1>
          <form action="" onSubmit={this.search}>
            <input type="search" className={style.searchField} tabIndex="0" />
            <button type="submit" className={style.btnSearch}>
              &#9906;
            </button>
          </form>
        </div>
      </header>
    );
  }
}

Header.defaultProps = {
  error: '',
  history: {},
};

Header.propTypes = {
  clearError: PropTypes.func.isRequired,
  error: PropTypes.string,
  history: PropTypes.objectOf(PropTypes.any),
};

export default Header;
