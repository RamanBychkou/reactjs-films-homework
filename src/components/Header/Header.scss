.header {
  width: 100%;
  padding: 0.9rem 0 0 0;
  background: linear-gradient(to bottom, rgba(0, 0, 0, 0.6), transparent);
  position: absolute;
  top: 0;
  overflow: hidden;
  left: 0;

  .container {
    width: 100%;
    max-width: 82rem;
    display: flex;
    flex-flow: row wrap;
    justify-content: space-between;
  }
}

.h1 {
  color: #fff;
  font-size: 1.56rem;
  letter-spacing: 2px;
  font-weight: 400;
}

.header form {
  position: relative;
}

.searchField {
  width: 16.8rem;
  height: 2rem;
  border: 1px solid #fff;
  border-radius: 4px;
  background: none;
  letter-spacing: 1px;
  color: #fff;
  text-align: left;
  cursor: pointer;
  outline: none;
  padding-left: 0.6rem;
}

.btnSearch {
  position: absolute;
  top: 0;
  right: 5px;
  font-size: 1.4rem;
  transform: rotate(-44deg);
  background: none;
  border: none;
  color: #fff;
  font-weight: 200;
  cursor: pointer;
}

@media screen and (max-width: 31.25rem) {
  .header .container {
    justify-content: center;
  }

  .h1 {
    margin-bottom: 1rem;
  }
}

@media screen and (min-width: 82rem) {
  .h1 {
    font-size: 3rem;
  }

  .container {
    margin: 0 auto;
  }
}
