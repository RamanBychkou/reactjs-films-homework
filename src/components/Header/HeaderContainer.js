import { connect } from 'react-redux';
import requestsFilms from '../../utils/requests';
import itemsReducer from '../../modules/Error/errorAction';
import Header from './Header';

const mapStateToProps = state => state;

export const mapStateToDispatch = dispatch => ({
  fetchSearchResults: query => dispatch(requestsFilms.fetchSearchResults(query)),
  clearError: boolean => dispatch(itemsReducer(boolean)),
});

export default connect(
  mapStateToProps,
  mapStateToDispatch,
)(Header);
