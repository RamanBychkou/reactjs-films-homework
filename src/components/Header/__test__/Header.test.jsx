import React from 'react';
import ReactDOM from 'react-dom';
import ReactTestRender from 'react-test-renderer';
import ReactTestUtils, { act } from 'react-dom/test-utils';
import { MemoryRouter as Router } from 'react-router-dom';
import Header from '../index';

describe('HeaderComponent', () => {
  const clearError = jest.fn();
  const state = {
    clearError,
    history: { push: jest.fn(), location: { path: '' } },
    query: { url: '', search: false },
  };
  test('Header renders correctly', () => {
    const result = ReactTestRender.create(<Header {...state} />);
    expect(result).toMatchSnapshot();
  });

  describe('Header: test search', () => {
    let container;
    beforeEach(() => {
      container = document.createElement('div');
      container.id = 'root';
    });

    it('Header: search test - correctly', () => {
      jest.spyOn(state.history, 'push');
      act(() => {
        document.body.appendChild(container);
        ReactDOM.render(
          <Router>
            <Header {...state} query={{ url: 'test', search: true }} />
          </Router>,
          container,
        );
      });
      const node = container.querySelector('form');
      ReactTestUtils.Simulate.submit(node);
      expect(state.history.push).toHaveBeenCalledTimes(1);
    });
  });
});
