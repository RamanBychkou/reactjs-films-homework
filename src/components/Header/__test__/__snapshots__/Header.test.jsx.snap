// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Header renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <h1
      className="h1"
    >
      FILMS
    </h1>
    <form
      action=""
      onSubmit={[Function]}
    >
      <input
        autoComplete="false"
        className="searchField"
        id="searchField"
        placeholder="search"
        tabIndex="0"
        type="search"
      />
      <input
        className="btnSearch"
        onClick={[Function]}
        type="button"
        value="âš²"
      />
    </form>
  </div>
</header>
`;
